x/derivatives/abci.go

```diff
@@ -4,12 +4,35 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/UnUniFi/chain/x/derivatives/keeper"
+	"github.com/UnUniFi/chain/x/derivatives/types"
 )
 
 func setPoolMarketCapSnapshot(ctx sdk.Context, k keeper.Keeper) {
+	// move on only if price is ready
+	if !IsPriceReady(ctx, k) {
+		return
+	}
+
 	k.SetPoolMarketCapSnapshot(ctx, ctx.BlockHeight(), k.GetPoolMarketCap(ctx))
 }
 
+func IsPriceReady(ctx sdk.Context, k keeper.Keeper) bool {
```

Comment:
This function should be implemented in keeper because it can be tested.

Reply:
oh, I forgot to attach it as keeper method.
thanks